src/esp/sensor/Sensor.cpp

```diff
@@ -42,14 +42,36 @@ void SensorSpec::sanityCheck() {
 
 Sensor::Sensor(scene::SceneNode& node, SensorSpec::ptr spec)
     : Magnum::SceneGraph::AbstractFeature3D{node}, spec_(std::move(spec)) {
+  CORRADE_ASSERT(node.children().first() == nullptr,
+                 "Sensor::Sensor(): Cannot attach a sensor to a non-LEAF node. "
+                 "The number of children of this node is not zero.", );
+  CORRADE_ASSERT(
+      node.getSceneNodeTags() & scene::SceneNodeTag::Leaf,
+      "Sensor::Sensor(): Cannot attach a sensor to a non-LEAF node.", );
   node.setType(scene::SceneNodeType::SENSOR);
   CORRADE_ASSERT(spec_,
                  "Sensor::Sensor(): Cannot initialize sensor. The "
                  "specification is null.", );
   spec_->sanityCheck();
+  node.getNodeSensorSuite().add(*this);
+  static_cast<scene::SceneNode&>(*node.parent())
+      .getNodeSensorSuite()
+      .add(*this);
+  node.getSubtreeSensorSuite().add(*this);
+  // Traverse up to root node and add sensor to every subtreeSensorSuite
```

Comment:
Is that possible to call the functions you wrote? e.g., `addSubtreeSensorsToAncestors`

Reply:
Thank you, good catch!